### PR TITLE
Feature/moses/create notification on comment

### PR DESCRIPTION
**Pull request by SapeleD3**
added notification for post and topics when Liked

### CODE REVIEW

File: functions/controllers/user/login.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ const login = async (req, res) => {
         const Token = await userData.user.getIdToken();
         return res.status(OK).json({ data: Token, status: success });
     } catch (err) {
+        console.log(err);
```
Comment (enyeInc) [inline functions/controllers/user/login.js:28]: remove this

File: functions/controllers/user/login.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ const login = async (req, res) => {
...
         return res.status(INTERNAL_SERVER_ERROR).json({
             message: somethingWentWrong,
             status: error,
```
Comment (enyeInc) [inline functions/controllers/user/login.js:31]: where is `error` coming from? I see `err` but no `error`

File: functions/controllers/user/signUp.js [hunk 3]
```diff
@@ -10,6 +12,8 @@ const {
 const { validateSignUpData } = require('../../util/validator');
 const { configConstants, status, message } = require('../../util/constants');
 
+dotenv.config();
```
Comment (enyeInc) [inline functions/controllers/user/signUp.js]: why are you loading the env files again. Only load is once at the most top level `index.js` file

File: functions/index.js [hunk 4]
```diff
@@ -1,14 +1,18 @@
+/* eslint-disable no-console */
+/* eslint-disable consistent-return */
+/* eslint-disable operator-linebreak */
```
Comment (enyeInc) [inline functions/index.js:3]: remove all of these eslint-disable

File: functions/index.js [hunk 5]
```diff
@@ -1,14 +1,18 @@
...
 const functions = require('firebase-functions');
 const swaggerUi = require('swagger-ui-express');
 const swaggerDocs = require('./swagger.json');
-
-dotenv.config();
+const { db } = require('./util/admin');
 
 const app = express();
 
+dotenv.config();
```
Comment (enyeInc) [inline functions/index.js:14]: This is the only place you should be loading these environment variables

File: functions/controllers/user/imageUpload.js [hunk 6]
```diff
@@ -5,13 +5,16 @@ const path = require('path');
...
 const { BAD_REQUEST, OK, INTERNAL_SERVER_ERROR } = require('http-status-codes');
+const dotenv = require('dotenv');
 const { admin } = require('../../util/admin');
 const { configConstants, status, message } = require('../../util/constants');
 
 const { error, success } = status;
 const { wrongFileSubmitted, somethingWentWrong, imageUpdateSucces } = message;
 const { IMAGE_JPEG, IMAGE_PNG } = configConstants;
 
+dotenv.config();
```
Comment (enyeInc) [inline functions/controllers/user/imageUpload.js]: remove

File: functions/index.js [hunk 7]
```diff
@@ -17,3 +21,106 @@ app.use('/docs', swaggerUi.serve, swaggerUi.setup(swaggerDocs));
...
+        postId: doc.id,
+        read: false,
+        recipient: doc.data().userName,
+        sender: snapshot.data().userName,
+        type: 'like',
+    });
+};
+exports.createNotificationOnLikeTopic = functions.firestore
+    .document('likes/{id}')
+    .onCreate(snapshot => {
```
Comment (enyeInc) [inline functions/index.js:48]: make this an async await function

File: functions/index.js [hunk 8]
```diff
@@ -17,3 +21,106 @@ app.use('/docs', swaggerUi.serve, swaggerUi.setup(swaggerDocs));
...
+                ) {
+                    return createLikeTopicNotification(snapshot, doc);
+                }
+            });
+    });
+
+exports.createNotificationOnLikePost = functions.firestore
+    .document('likes/{id}')
+    .onCreate(snapshot => {
+        db.doc(`/posts/${snapshot.data().postId}`)
```
Comment (enyeInc) [inline functions/index.js:64]: make this an async await function

File: functions/index.js [hunk 9]
```diff
@@ -17,3 +21,106 @@ app.use('/docs', swaggerUi.serve, swaggerUi.setup(swaggerDocs));
...
+    .onDelete(snapshot => {
+        db.doc(`/notifications/${snapshot.id}`)
+            .delete()
+            .catch(err => console.error(err));
+    });
+
+exports.createNotificationOnCommentPost = functions.firestore
+    .document('comments/{id}')
+    .onCreate(snapshot => {
+        db.doc(`/posts/${snapshot.data().postId}`)
```
Comment (enyeInc) [inline functions/index.js:87]: make this an async await function

File: functions/index.js [hunk 10]
```diff
@@ -17,3 +21,106 @@ app.use('/docs', swaggerUi.serve, swaggerUi.setup(swaggerDocs));
...
+                        sender: snapshot.data().userName,
+                        type: 'comment',
+                    });
+                }
+            });
+    });
+
+exports.createNotificationOnCommentTopic = functions.firestore
+    .document('comments/{id}')
+    .onCreate(snapshot => {
```
Comment (enyeInc) [inline functions/index.js:108]: make this an async await function

File: functions/util/admin.js [hunk 11]
```diff
@@ -1,8 +1,11 @@
 const admin = require('firebase-admin');
 const firebase = require('firebase');
+const dotenv = require('dotenv');
 // const serviceAccount = require('../serviceAccount.json');
 require('firebase/firestore');
 
+dotenv.config();
```
Comment (enyeInc) [inline functions/util/admin.js:6]: remove

File: functions/util/admin.js [hunk 12]
```diff
@@ -1,8 +1,11 @@
 const admin = require('firebase-admin');
 const firebase = require('firebase');
+const dotenv = require('dotenv');
 // const serviceAccount = require('../serviceAccount.json');
```
Comment (enyeInc) [inline functions/util/admin.js]: remove unused code

**Review by enyeInc — CHANGES_REQUESTED** [review]
Code needs to be cleaned up

File: functions/controllers/user/login.js (hunk 2 above)
Comment (SapeleD3) [reply functions/controllers/user/login.js:31]: error is coming from my constant folder as a string
sir tosins recommendation earlier

**Pull request merged**